### ISSUE TITLE
Autocorrected safe navigation violates new navigation chain rules

### ISSUE THREAD

**Issue by jamesoneill997** [rubocop/rubocop#13423]
## Expected behavior
Autocorrect should not introduce futher errors. Either autocorrecting now expands safe navigation chains, ignores/encourages object checks, or the offence isn't autocorrectable). 

## Actual behavior
Was getting a warning from rubocop upon updating to 1.68.0 that the navigation chains were too long. Code that was caught: 
```
      foo&.bar&.abc&.xyz&.name
```

I amended this by checking the base object existed before querying fields:

```
      foo & foo.bar.abc&.xyz&.name
```

I re-ran rubocop to find that it found a correctable offence:
```
app/path/to/file.rb:42:7: C: [Corrected] Style/SafeNavigation: Use safe navigation (&.) instead of checking if an object exists before calling the method.
      foo && foo.bar.abx&.xyz&.name
``` 
      
 Updated code after autocorrect reverted it back to the original code that I had with the navigation chain:
```
       foo&.bar&.abc&.xyz&.name
```

Linking the new feature for visibility:
https://github.com/rubocop/rubocop/issues/13171

## Steps to reproduce the problem

Create a block of code where you check for the existance of a base object before chaining.
Run rubocop, it should detect an offence
Run rubocop with --autocorrect-all and it will revert to having a chain that breaches the safe navigation chain length rule.

## RuboCop version

```
jamesoneill@Mac:~/src/my_app(my_branch⚡) » rubocop -v                                                                                     1 ↵
1.68.0
```

**Comment by dvandersluis** [comment]
I think this is the same issue as #13421. @koic maybe `Style/SafeNavigation` should report but not autocorrect if the resulting chain is longer than the configuration for `Style/SafeNavigationChainLength`?

**Comment by koic** [comment]
This appears to be a bug in `Style/SafeNavigation`. I've reopened #13421 and integrate it there.

**Comment by jamesoneill997** [comment]
Thanks very much @koic & @dvandersluis. Really appreciate the quick response 💪
